### PR TITLE
Conversion enu for webots ros

### PR DESCRIPTION
**Pull request by BenjaminHug8**
This PR has the goal to convert worlds and cpp files in order to be compatible with 2021c.

- [x] Convert worlds
- [x] Convert scripts

### CODE REVIEW

**Comment by BenjaminHug8** [comment]
@ad-daniel May I ask you to distribute the review of this PR please?

File: worlds/keyboard_teleop.wbt [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
+  orientation -0.30726263682167737 0.11409240523127144 0.9447606019949887 2.4520204012157247
+  position 7.080240838628136 -6.127864336042442 6.717421941146577
 }
 TexturedBackground {
 }
 TexturedBackgroundLight {
 }
 RectangleArena {
-  rotation 1 0 0 1.5707963267948966
+  rotation 1 0 0 0
```
Comment (joachimhgg) [inline worlds/keyboard_teleop.wbt:13]:
```suggestion
```

File: worlds/pioneer3at.wbt [hunk 2]
```diff
@@ -9,451 +9,432 @@ WorldInfo {
...
   translation 7.57085 2.42019 -4.52971e-14
-  rotation 1 0 0 1.5707963267948966
   name "carpet(1)"
 }
 DEF EXTERNAL_WALL Wall {
-  translation 0 30 0
-  rotation 1 0 0 1.5707963267948966
+  translation 0 30.1 1.2
+  rotation 1 0 0 1.5708
   size 22.4 2.4 0.2
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:36]:
```suggestion
  translation 0 0 30
  rotation 0 1 0 1.5708
  size 0.2 22.4 2.4
```
something like this. Need to be done for the external walls.

File: worlds/pioneer3at.wbt [hunk 3]
```diff
@@ -9,451 +9,432 @@ WorldInfo {
...
 }
 TrafficCone {
   translation -22.831 13.9376 9.166e-13
-  rotation 1 0 0 1.5707963267948966
   name "traffic cone(6)"
 }
 TrafficCone {
   translation -18.9491 14.1135 4.47642e-13
-  rotation -0.577350661639742 0.577350072964468 0.577350072964468 4.188790793461465
+  rotation -7.19235e-09 7.19235e-09 -1 1.5708
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:399]:
```suggestion
  rotation 0 0 -1 1.5708
```

File: worlds/pioneer3at.wbt [hunk 4]
```diff
@@ -9,451 +9,432 @@ WorldInfo {
...
   name "traffic cone(7)"
 }
 TrafficCone {
   translation -13.7591 13.8483 -1.42109e-14
-  rotation -0.476905294946303 0.621514818669748 0.621514818669748 4.031596129182271
+  rotation -5.51888e-09 7.19235e-09 -1 1.8326
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:404]:
```suggestion
  rotation 0 0 -1 1.8326
```

File: worlds/pioneer3at.wbt [hunk 5]
```diff
@@ -471,15 +452,15 @@ DEF KITCHEN Transform {
...
+        translation 0 -0.3 0
         handleLength 0.1
         handleRadius 0.008
         handleColor 0.427451 0.513725 0.533333
       }
     }
     Cabinet {
       translation -1.65 0.15 0.11
-      rotation 0 1 0 1.5708
+      rotation 1 -1.72486e-06 -6.88391e-07 -1.5707953071795862
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:463]:
```suggestion
      rotation 1 0 0 -1.5707953071795862
```

File: worlds/pioneer3at.wbt [hunk 6]
```diff
@@ -548,15 +530,15 @@ DEF KITCHEN Transform {
...
+        translation 0 -0.2 0
         handleLength 0.1
         handleRadius 0.008
         handleColor 0.427451 0.513725 0.533333
       }
     }
     Cabinet {
       translation -1.48 0.15 -0.79
-      rotation 0 1 0 1.5708
+      rotation 1 -1.72486e-06 -6.88391e-07 -1.5707953071795862
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:541]:
```suggestion
      rotation 1 0 0 -1.5707953071795862
```

File: worlds/pioneer3at.wbt [hunk 7]
```diff
@@ -700,20 +683,20 @@ DEF KITCHEN Transform {
     }
     Sink {
       translation -1.33 0.87 -0.78
-      rotation 0 1 0 4.71239
+      rotation -0.9999999999944038 8.636279999951669e-07 3.2321499999819125e-06 1.57079
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:686]:
```suggestion
      rotation -1 0 0 1.57079
```

File: worlds/pioneer3at.wbt [hunk 8]
```diff
@@ -747,61 +730,60 @@ DEF PIONEER_3AT Pioneer3at {
...
 Table {
   translation 22.5 -21.6 0
-  rotation 1 0 0 1.5707963267948966
-  size 4.2 0.74 1
+  size 4.2 1 0.74
 }
 Chair {
   translation 23.2032 -22.0662 0
-  rotation -0.577350661639742 -0.577350072964468 -0.577350072964468 4.188790793461465
+  rotation -7.19235e-09 -7.19235e-09 1 1.5708
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:737]:
```suggestion
  rotation 0 0 1 1.5708
```

File: worlds/pioneer3at.wbt [hunk 9]
```diff
@@ -747,61 +730,60 @@ DEF PIONEER_3AT Pioneer3at {
...
 }
 Chair {
   translation 23.2032 -21.2162 0
-  rotation -0.577350661639742 0.577350072964468 0.577350072964468 4.188790793461465
+  rotation -7.19235e-09 7.19235e-09 -1 1.5708
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:741]:
```suggestion
  rotation 0 0 -1 1.5708
```

File: worlds/pioneer3at.wbt [hunk 10]
```diff
@@ -747,61 +730,60 @@ DEF PIONEER_3AT Pioneer3at {
...
   name "chair(1)"
 }
 Chair {
   translation 22.4132 -22.0562 0
-  rotation -0.577350661639742 -0.577350072964468 -0.577350072964468 4.188790793461465
+  rotation -7.19235e-09 -7.19235e-09 1 1.5708
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:746]:
```suggestion
  rotation 0 0 1 1.5708
```

File: worlds/pioneer3at.wbt [hunk 11]
```diff
@@ -747,61 +730,60 @@ DEF PIONEER_3AT Pioneer3at {
...
   name "chair(2)"
 }
 Chair {
   translation 22.4132 -21.2162 0
-  rotation -0.577350661639742 0.577350072964468 0.577350072964468 4.188790793461465
+  rotation -7.19235e-09 7.19235e-09 -1 1.5708
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:751]:
```suggestion
  rotation 0 0 -1 1.5708
```

File: worlds/pioneer3at.wbt [hunk 12]
```diff
@@ -931,802 +912,885 @@ DEF TREE Solid {
...
-      name "desk(28)"
+      rotation 0.5773529358523396 0.5773499358526729 0.5773479358528951 -2.094395307179586
+      name "desk(14)"
     }
     Desk {
       translation 22.9277 7.10543e-15 -25.4837
-      rotation 0 1 0 6.28319
-      name "desk(29)"
+      rotation 0.5773529358523396 0.5773499358526729 0.5773479358528951 -2.094395307179586
+      name "desk(30)"
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:1584]:
```suggestion
      name "desk(15)"
```

**Review by joachimhgg — CHANGES_REQUESTED** [review]
Thanks ! Just a few minors changes.

-catch_the_birds:
    - wrong orientation of the body of the bird
- complete_test:
     - an extra cone and a person in robots have been added.
     - the robot and the cube are not FLU (Z up)
     - the RADAR_TARGET is not at the good location
- epuck_line:
    - obstacles are not FLU (change box size from size 1 2 3 to size 3 1 2 )
- pioneer3at:
    - EXTERNAL_WALL INTERNAL_WALL are not flu. (change box size + rotation)
    - rotations need to be cleaned. You can use the script `scripts/converter/convert_nue_to_enu_rub.py` and change the mode by 'clean' to clean all the rotations.
- tiago:
    - e-puck is not at the good location
    - worktops and walls are not flu
    - rotations need to be cleaned

**Comment by BenjaminHug8** [comment]
I changed the Kitchen in flu in `pioneer3at.wbt`.

File: worlds/catch_the_bird.wbt [hunk 13]
```diff
@@ -229,15 +227,28 @@ Solid {
...
+              }
+            }
+          ]
+          name "body_center"
+          boundingObject Capsule {
             height 0.07
             radius 0.02
+            subdivision 16
+          }
+          physics Physics {
```
Comment (joachimhgg) [inline worlds/catch_the_bird.wbt:251]: image

you added an extra bounding object, physics node and name

File: worlds/complete_test.wbt [hunk 14]
```diff
@@ -76,11 +74,14 @@ DEF RADAR_TARGET Solid {
 }
 Robot {
   translation 0 0 0.1
-  rotation 1 0 0 1.5707963267948966
+  rotation 1 0 0 0
```
Comment (joachimhgg) [inline worlds/complete_test.wbt:77]:
```suggestion
```

File: worlds/complete_test.wbt [hunk 15]
```diff
@@ -276,7 +275,8 @@ Robot {
       maxRange 2
     }
     CharacterSkin {
-      translation -0.35 -0.1 0
+      translation -0.35 0 -0.1
+      rotation 0.707107781185531 1.695249475405131e-09 0.70710578118615 -3.1415853071795863
```
Comment (joachimhgg) [inline worlds/complete_test.wbt:279]:
```suggestion
      rotation 0.707107781185531 0 0.70710578118615 -3.1415853071795863
```

File: worlds/pioneer3at.wbt [hunk 16]
```diff
@@ -930,803 +895,886 @@ DEF TREE Solid {
...
 PottedTree {
-  translation 17.1715 -20.9225 3.55271e-15
-  rotation 1 0 0 1.5707963267948966
+  translation 17.1715 -20.9225 0
   name "potted tree(2)"
 }
 Sofa {
   translation 27.9791 -28.4787 0
-  rotation 0.281086281604685 0.678598004084028 0.678598004084027 2.593561198933153
+  rotation 0 0 -1 -2.356185307179586
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:910]:
```suggestion
  rotation 0 0 1 2.356185307179586
```

File: worlds/pioneer3at.wbt [hunk 17]
```diff
@@ -930,803 +895,886 @@ DEF TREE Solid {
...
-      name "desk(14)"
+      translation 28.2877 0 -16.1837
+      rotation 0.5773529358523396 0.5773499358526729 0.5773479358528951 -2.094395307179586
+      name "desk(30)"
     }
   ]
 }
 DEF OFFICES Transform {
-  rotation 9.38185669e-07 0.707106781186236 0.707106781186236 3.141590777218456
+  rotation -1 0 0 -1.5707953071795862
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:1243]:
```suggestion
  rotation 1 0 0 1.5707953071795862
```

File: worlds/pioneer3at.wbt [hunk 18]
```diff
@@ -1802,8 +1850,8 @@ StraightStairs {
   ]
 }
 StraightStairs {
-  translation 0.76 19 0
-  rotation 0.577348855372322 0.577350976096979 0.577350976096979 2.094397223120449
+  translation 0.76 19.0 0
+  rotation 0 0 -1 -1.5707953071795862
```
Comment (joachimhgg) [inline worlds/pioneer3at.wbt:1854]:
```suggestion
  rotation 0 0 1 1.5707953071795862
```

File: worlds/tiago.wbt [hunk 19]
```diff
@@ -69,42 +68,42 @@ DEF ROOF Solid {
...
-  rotation 1 0 0 1.5707963267948966
+  rotation 0 0 1 1.5707963267948966
   name "wall 4"
-  size 1 2.4 0.3
+  size 0.3 1 2.4
 }
 Door {
   translation 4.95 -9.9 0
-  rotation 9.38185669e-07 0.707106781186236 0.707106781186236 3.141590777218456
+  rotation 0 0 -1 -1.5707853071795865
```
Comment (joachimhgg) [inline worlds/tiago.wbt:104]:
```suggestion
  rotation 0 0 1 1.5707853071795865
```

File: worlds/tiago.wbt [hunk 20]
```diff
@@ -280,196 +267,177 @@ Cabinet {
...
 }
 Table {
   translation 4.94438 -1.07424 0
-  rotation 0.577348855372322 0.577350976096979 0.577350976096979 2.094397223120449
   name "table(1)"
 }
 Chair {
   translation 5.41278 -1.46063 -0.00224453
-  rotation 3.2757990828e-05 0.707106780807154 0.707106780807154 3.141527137608161
+  rotation 0 0 -1 -3.141495307179586
```
Comment (joachimhgg) [inline worlds/tiago.wbt:309]:
```suggestion
  rotation 0 0 1 3.141495307179586
```

File: worlds/tiago.wbt [hunk 21]
```diff
@@ -280,196 +267,177 @@ Cabinet {
...
     centerOfMass [
-      0 0.45 0
+      0 0 0.45
     ]
   }
 }
 Chair {
   translation 5.38086 -0.706899 -0.00224453
-  rotation 3.2757990828e-05 0.707106780807154 0.707106780807154 3.141527137608161
+  rotation 0 0 -1 -3.141495307179586
```
Comment (joachimhgg) [inline worlds/tiago.wbt:340]:
```suggestion
  rotation 0 0 1 3.141495307179586
```

File: worlds/tiago.wbt [hunk 22]
```diff
@@ -280,196 +267,177 @@ Cabinet {
...
 }
 Sofa {
   translation 0.804228 -7.05325 0
-  rotation 1 0 0 1.5707963267948966
   color 0.13333333333333333 0.13333333333333333 0.13333333333333333
 }
 Sofa {
   translation 2.49729 -8.95734 0
-  rotation 0.577348855372322 0.577350976096979 0.577350976096979 2.094397223120449
+  rotation 0 0 -1 -1.5707953071795862
```
Comment (joachimhgg) [inline worlds/tiago.wbt:400]:
```suggestion
  rotation 0 0 1 1.5707953071795862
```

File: worlds/tiago.wbt [hunk 23]
```diff
@@ -494,48 +462,46 @@ Cabinet {
...
 }
 PottedTree {
   translation 6.08049 -4.51701 0
-  rotation 1 0 0 1.5707963267948966
+  rotation 0 0 1 1.5708
 }
 PortraitPainting {
   translation 2.47 -9.73 1.6
-  rotation 9.38185669e-07 0.707106781186236 0.707106781186236 3.141590777218456
+  rotation 0 0 -1 -1.5707953071795862
```
Comment (joachimhgg) [inline worlds/tiago.wbt:494]:
```suggestion
  rotation 0 0 1 1.5707953071795862
```

**Review by joachimhgg — CHANGES_REQUESTED** [review]
Thanks. Just a few more comments.
- catch_the_birds: 
  - wrong bounding box
- complete_test:
image
2021b version:
image

- the robot miss a -pi/2 on z to be FLU.
- the Solid (the box) is not FLU.
- you can delete the rotation on the RADAR_TARGET to be FLU

**Comment by BenjaminHug8** [comment]
For the `catch_the_bird`, due to the change of orientation of the cylinders I have no choice than add `Transform` nodes.

**Comment by joachimhgg** [comment]
complete_test: The range finder needs to be turn as well

**Review by joachimhgg — APPROVED** [review]

**Pull request merged**